src/analysis/convex/function.lean

```diff
@@ -104,6 +104,51 @@ lemma strict_concave_on.subset {t : set E} (hf : strict_concave_on 𝕜 t f) (hs
   strict_concave_on 𝕜 s f :=
 ⟨hs, λ x y hx hy, hf.2 (hst hx) (hst hy)⟩
 
+section composition
+
+variables [has_smul 𝕜 F] [ordered_add_comm_monoid F]
```

Comment:
The linter says that you also have an `[add_comm_monoid F]` instance (probably, from an earlier `variables` command). You should close the section that has this variable before starting the section about compositions. Or move some code into yet another subsection.

Reply:
I switched back to introducing a new variable as the ordered_add_comm_monoid. Otherwise I would have to close almost all of the created sections and then recreate them right after.